R/teal_data_utils.R

```diff
@@ -42,35 +42,37 @@ NULL
 .subset_teal_data <- function(data, datanames) {
   checkmate::assert_class(data, "teal_data")
   checkmate::assert_class(datanames, "character")
-  datanames_corrected <- intersect(datanames, ls(data@env))
-  dataname_corrected_with_raw <- intersect(c(datanames, sprintf("%s._raw_", datanames)), ls(data@env))
-
+  datanames_corrected <- intersect(datanames, .teal_data_ls(data))
+  datanames_corrected_with_raw <- c(datanames_corrected, ".raw_data")
   if (!length(datanames)) {
     return(teal_data())
   }
```

Comment:
this should be run before `datanames_corrected`

Reply:
addressed